FinSearchUnified/Tests/BusinessLogic/Models/FindologicArticleModelTest.php

```diff
@@ -247,4 +247,111 @@ public function articlesWithAttributeProvider()
             ]
         ];
     }
+
+    /**
+     * @dataProvider articleSEOUrlProvider
+     *
+     * @param array $articleConfiguration
+     * @param string $expectedUrl
+     *
+     * @throws ReflectionException
+     */
+    public function testArticleWithSEOUrl(array $articleConfiguration, $expectedUrl)
+    {
+        $baseCategory = new Category();
+        $baseCategory->setId(5);
+
+        $articleFromConfiguration = $this->createTestProduct($articleConfiguration);
+
+        $shop = Manager::getResource('Shop')->getRepository()->find(1);
+        $shop->registerResources();
+
+        Shopware()->Modules()->RewriteTable()->sInsertUrl(
+            'sViewport=detail&sArticle=' . $articleFromConfiguration->getId(),
+            $articleFromConfiguration->getName() . '/'
+        );
+
+        $findologicArticle = $this->articleFactory->create(
+            $articleFromConfiguration,
+            'ABCD0815',
+            [],
+            [],
+            $baseCategory
+        );
+
+        $xmlArticle = $findologicArticle->getXmlRepresentation();
+
+        $reflector = new ReflectionClass(Item::class);
+        $properties = $reflector->getProperty('url');
+        $properties->setAccessible(true);
+        $values = $properties->getValue($xmlArticle);
+        $actualUrl = current($values->getValues());
+
+        // Remove the host from the URL for assertion
+        $actualUrl = explode(Shopware()->Shop()->getHost(), $actualUrl)[1];
```

Comment:
I don't have a good feeling about this. Since we are adding the shop's URL to the product URL it might happen that we are doing something wrong. In that case we would never notice it because we don't consider the shop's hostname in the test.
What if we added `Shopware()->Shop()->getHost()` to `$expectedUrl` instead of removing it from `$actualUrl`?